sql/src/main/java/io/crate/protocols/postgres/BulkPortal.java

```diff
@@ -136,49 +135,32 @@ public void execute(ResultReceiver resultReceiver, int maxRows) {
 
     private CompletableFuture<Void> executeBulk(Executor executor, Plan plan, final UUID jobId,
                                                 final StatsTables statsTables) {
-        final CompletableFuture<Void> bulkCompleteFuture = new CompletableFuture<>();
         List<CompletableFuture<Long>> futures = executor.executeBulk(plan);
+        CompletableFuture<Void> allFutures = CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
```

Comment:
Use the futures.size in the array you pass in to futures.toArray to avoid another of the array being created (ie futures.toArray(new CompletableFuture[futures.size()] )